### PR TITLE
[Merged by Bors] - chore(Algebra.Order.Ring.Rat): split into unbundled and bundled ordered algebra results

### PR DESCRIPTION
**Pull request by mattrobball**
We split off all but the `LinearOrderedCommRing` and the inferred `OrderedAddCommMonoid` instances on `Rat` into `Algebra.Order.Ring.Unbundled.Rat` as these results do not require bundled ordered algebra classes. 

---

- [x] depends on: #15069 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Ring/Unbundled/Rat.lean [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+import Mathlib.Algebra.Order.Group.Unbundled.Basic
+import Mathlib.Algebra.Ring.Rat
+import Mathlib.Init.Data.Int.Order
+
+/-!
+# The rational numbers possess a linear order
+
+This file constructs the order on `ℚ` and proves various facts relating the order to
+ring structure on `ℚ`.
+
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Ring/Unbundled/Rat.lean:16]: Can you allude to the `Unbundled` name in this docstring? Perhaps "This file only provides the unbundled `CovariantClass` instance; the bundled `LInearOrderedField` instance is left to a later file".

Comment (mattrobball) [reply Mathlib/Algebra/Order/Ring/Unbundled/Rat.lean:18]: Done

**Comment by j-loreaux** [comment]
I think it's a good change, but I'm curious if there are performance benefits.

!bench

**Comment by mattrobball** [comment]
!bench

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**